---
int32 state
uint8 id
